src/UnoFlipController.java

```diff
@@ -75,26 +75,29 @@ public void actionPerformed(ActionEvent e) {
             String[] numPlayerOptions = {"RED", "BLUE", "YELLOW", "GREEN"};
 
             JComboBox<String> comboBox = new JComboBox<>(numPlayerOptions);
-
-            int result = JOptionPane.showOptionDialog(
-                    null,
-                    comboBox,
-                    "Select colour:",
-                    JOptionPane.DEFAULT_OPTION,
-                    JOptionPane.PLAIN_MESSAGE,
-                    null,
-                    null,
-                    null);
-
-            Card.Colour colour;
-            if (result == JOptionPane.OK_OPTION) {
-                String option = (String) comboBox.getSelectedItem();
-                colour = Card.Colour.valueOf(option);
-                this.model.setCurrentColour(colour);
-                this.model.notifyViews();
-            } else {
-                System.exit(0); // absolutely must select an option, otherwise, do not start the game
+            while(true){
+                int result = JOptionPane.showOptionDialog(
+                        null,
+                        comboBox,
+                        "Select colour:",
+                        JOptionPane.DEFAULT_OPTION,
+                        JOptionPane.PLAIN_MESSAGE,
+                        null,
+                        null,
+                        null);
+
+                Card.Colour colour;
```

Comment:
Don't need to allocate memory for this variable here. Its scope is only the if-block, so allocate it only if you get inside the if-block. Otherwise, it's being allocated for no reason